Java OOP advanced

VladGural opened the pull request:
Created classes of 5 figures supper class Figure Interface Drawable classes of Supplier and logic in Main method

MaliukDaria commented on src/main/java/figeres/Figure.java:
```diff
@@ -0,0 +1,19 @@
+        this.color = color;
+    }
+
+    public abstract double getArea();
```
let's move this method into some `AreaCalculator` interface

MaliukDaria commented on src/main/java/suppliers/ColorSupplier.java:
```diff
@@ -0,0 +1,29 @@
+import java.util.Random;
+
+public class ColorSupplier {
+    private String[] colours = {"white", "black", "red", "orange",
+            "yellow", "green", "blue", "purple"};
```
let's make colors array constant field?

VladGural commented:
Created Interface AreaCalculator In class ColorSupplier 
Variable private String[] colours was changed to public static final String[] colours
Changed package's name from figeres to figures

MaliukDaria commented on src/main/java/suppliers/ColorSupplier.java:
```diff
@@ -0,0 +1,21 @@
+import java.util.Random;
+
+public class ColorSupplier {
+    public static final String[] colours = {"white", "black", "red", "orange",
+            "yellow", "green", "blue", "purple"};
```
what about constant naming? You can see examples [here](https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names)

MaliukDaria commented on src/main/java/suppliers/FigureSupplier.java:
```diff
@@ -0,0 +1,61 @@
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Random getRandom() {
+        return random;
+    }
+
+    public void setRandom(Random random) {
+        this.random = random;
+    }
+
+    public ColorSupplier getColorSupplier() {
+        return colorSupplier;
+    }
+
+    public void setColorSupplier(ColorSupplier colorSupplier) {
+        this.colorSupplier = colorSupplier;
+    }
```
I don't think we need this

MaliukDaria commented on src/main/java/suppliers/FigureSupplier.java:
```diff
@@ -0,0 +1,61 @@
+        Figure figure;
+        switch (figureNumber) {
+            case 0 :
+                figure = new Square(colorSupplier.getRandomColor(), random.nextInt(20));
```
let's move all magic numbers in constants

MaliukDaria replied:
В условиях case можно оставить цифры. Мой предыдущий комментарий касался цифры `20`. Давай вернем условия case как было, а цифру 20 вынесем в константу

MaliukDaria commented on src/main/java/suppliers/ColorSupplier.java:
```diff
@@ -0,0 +1,21 @@
+            "yellow", "green", "blue", "purple"};
+    private Random random = new Random();
+
+    public Random getRandom() {
+        return random;
+    }
+
+    public void setRandom(Random random) {
+        this.random = random;
+    }
```
I don't think we need it

VladGural commented:
Deleted unuseful getters and setters in classes ColorSupplier and FigureSupplier
Changed constant name from colours to COLORS
Created constants of magic numbers

MaliukDaria commented on src/main/java/core/basesyntax/MainApp.java:
```diff
@@ -0,0 +1,19 @@
+public class MainApp {
+    public static void main(String[] args) {
+        Random random = new Random();
+        Figure[] figures = new Figure[random.nextInt(10)];
```
Давай цифру `10` вынесем в константу

VladGural commented:
Revert "created constants of magic numbers" …
Created constants MAX_NUMBER_OF_FIGURES and MAX_SIZE_OF_FIGURE

MaliukDaria commented on src/main/java/core/basesyntax/MainApp.java:
```diff
@@ -0,0 +1,21 @@
+import suppliers.FigureSupplier;
+
+public class MainApp {
+    static final int MAX_NUMBER_OF_FIGURES = 10;
```
what about access modifiers?

MaliukDaria commented on src/main/java/suppliers/ColorSupplier.java:
```diff
@@ -0,0 +1,13 @@
+import java.util.Random;
+
+public class ColorSupplier {
+    static final String[] COLORS = {"white", "black", "red", "orange",
```
what about access modifiers?

MaliukDaria commented on src/main/java/suppliers/FigureSupplier.java:
```diff
@@ -0,0 +1,49 @@
+import java.util.Random;
+
+public class FigureSupplier {
+    static final int FIGURE_COUNT = 5;
+    static final int MAX_SIZE_OF_FIGURE = 20;
```
access modifiers

VladGural commented:
Changed access modifiers all constants to private
